docs/tests.md

```diff
@@ -0,0 +1,27 @@
+## Location of tests
+
+Take the Write API. It has:
+
+1. Unit tests in pkg/server/commands/write_test.go. These mock all the external dependencies. The tests here should test all business logic that is independent of the external dependencies.
+2. **Exported** integration tests in pkg/server/test/write.go. These take in a `datastore` dependency. This is so that anyone that writes their own implementation of the datastore can run these tests by doing
```

Comment:
I think it's also worth explicitly calling out (in another bullet point) that `pkg/server/test` is a suite of server conformance tests that are publicly exported so that external projects can test the Server with a provided datastore implementation.